src/app/InteractionModelEngine.cpp

```diff
@@ -530,10 +532,12 @@ CHIP_ERROR InteractionModelEngine::OnUnsolicitedReportData(Messaging::ExchangeCo
             continue;
         }
 
-        return readClient->OnUnsolicitedReportData(apExchangeContext, std::move(aPayload));
+        VerifyOrReturnError(readClient->OnUnsolicitedReportData(apExchangeContext, std::move(aPayload)) == CHIP_NO_ERROR,
```

Comment:
So... I think this is actually buggy.

Let's say we call `OnUnsolicitedReportData`, it lands in `ProcessReportData`, that hits an error.

We're going to send a status response at the end of `ProcessReportData` and return the error.  `OnUnsolicitedReportData` will return the error to here.  This code will return `InvalidAction`, which will get propagated to the caller, which will send _another_ StatusResponse, no?  Or am I missing something that makes this work out OK?

I think `OnUnsolicitedReportData` should just return void...

Reply:
I have changed ProcessReportData where it would only send status report with success when the processing is successful. When there is error, the caller would take care of status report with error which would not expect any response

Reply:
Ah, I see.  So in the success case we send in `ProcessReportData`, and in the failure cases we send the status either here or in OnMessageReceived?  That works, ok.